src/Data/List/Properties.agda

```diff
@@ -764,84 +764,152 @@ module _ {P : A → Set p} (P? : Decidable P) where
   ...  | no ¬Px = P.cong (Prod.map id (x ∷_)) (partition-defn xs)
 
 ------------------------------------------------------------------------
--- reverse
+-- reverse is an involution in the list monoid
 
 module _ {a} {A : Set a} where
 
   open FunctionProperties {A = List A} _≡_
+  open P.≡-Reasoning
 
-  unfold-reverse : ∀ (x : A) xs → reverse (x ∷ xs) ≡ reverse xs ∷ʳ x
-  unfold-reverse x xs = helper [ x ] xs
-    where
-    open P.≡-Reasoning
-    helper : (xs ys : List A) → foldl (flip _∷_) xs ys ≡ reverse ys ++ xs
-    helper xs []       = refl
-    helper xs (y ∷ ys) = begin
-      foldl (flip _∷_) (y ∷ xs) ys  ≡⟨ helper (y ∷ xs) ys ⟩
-      reverse ys ++ y ∷ xs          ≡⟨ P.sym (++-assoc (reverse ys) _ _) ⟩
-      (reverse ys ∷ʳ y) ++ xs       ≡⟨ P.sym $ P.cong (_++ xs) (unfold-reverse y ys) ⟩
-      reverse (y ∷ ys) ++ xs        ∎
+  -- Defining property of reverse-append _ʳ++_.
 
-  reverse-++-commute : (xs ys : List A) →
-                       reverse (xs ++ ys) ≡ reverse ys ++ reverse xs
-  reverse-++-commute []       ys = P.sym (++-identityʳ _)
-  reverse-++-commute (x ∷ xs) ys = begin
-    reverse (x ∷ xs ++ ys)               ≡⟨ unfold-reverse x (xs ++ ys) ⟩
-    reverse (xs ++ ys) ++ [ x ]          ≡⟨ P.cong (_++ [ x ]) (reverse-++-commute xs ys) ⟩
-    (reverse ys ++ reverse xs) ++ [ x ]  ≡⟨ ++-assoc (reverse ys) _ _ ⟩
-    reverse ys ++ (reverse xs ++ [ x ])  ≡⟨ P.sym $ P.cong (reverse ys ++_) (unfold-reverse x xs) ⟩
-    reverse ys ++ reverse (x ∷ xs)       ∎
-    where open P.≡-Reasoning
+  ʳ++=reverse++ : ∀ (xs : List A) {ys} → xs ʳ++ ys ≡ reverse xs ++ ys
```

Comment:
I'm aware it's not quite as meaningful, but to be consistent with the naming conventions used in the rest of the file we should probably call this `ʳ++-defn`

Reply:
Ok!